src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.LockTest.Unix.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Xunit;
+
+public class FileSystemWatcherLockTest
+{
+    [Fact]
+    public static void FileSystemWatcher_WatchLocksWatchedDirectory()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var fsw = new FileSystemWatcher())
+        {
+            fsw.Path = Path.GetFullPath(dir.Path);
+            fsw.EnableRaisingEvents = true;
+
+            Microsoft.Win32.SafeHandles.SafeFileHandle handle = Interop.Sys.Open(fsw.Path, Interop.Sys.OpenFlags.O_RDONLY, 0);
+            Assert.False(handle.IsInvalid); // We should always be able to open a handle
+            Assert.True(Interop.Sys.FLock(handle, Interop.Sys.LockOperations.LOCK_EX | Interop.Sys.LockOperations.LOCK_NB) < 0);
+
+            Interop.ErrorInfo err = Interop.Sys.GetLastErrorInfo();
+            Assert.True(err.Error == Interop.Error.EWOULDBLOCK); // validate it is a sharing violation
+        }
+    }
+}
```

Comment:
I found this just from experience with the Win32 APIs and wondering if the OS X and Linux ones worked the same way. This change is more to try and be a good citizen on the system...I think we should use locking in the underlying Unix FileSystem classes if we would on Windows, but that's a different change. I agree that this can't actually prevent I/O, but it does help us play nice with other applications on the system that are also good citizens.